doc/man3/ASN1_TIME_set.pod

```diff
@@ -32,7 +34,16 @@ and returned.
 
 ASN1_TIME_set_string() sets ASN1_TIME structure B<s> to the time
 represented by string B<str> which must be in appropriate ASN.1 time
-format (for example YYMMDDHHMMSSZ or YYYYMMDDHHMMSSZ).
+format (for example YYMMDDHHMMSSZ or YYYYMMDDHHMMSSZ). If B<s> is NULL
+this function performs a format check on B<str> only.
+
+ASN1_TIME_set_string_X509() is more 'strict' than ASN1_TIME_set_string()
+while checking the string B<str>'s format against RFC 5280's rules. For
+instance, SS and Z must be present in a UTCTime format and fractional
+seconds are not allowed in a GeneralizedTime format, and in both format,
+time zones provided by + or - are not allowed. Besides, if the year presented
+by B<str> is between 1950 (included) and 2049, the ASN1_TIME structure B<s>
+is set in UTCTime, otherwise it will be set to GeneralizedTime.
```

Comment:
I think we don't need to document in detail what it all checks, just that it's restricts it to what RFC5280 allows. So maybe it should say that it only allows YYMMDDHHMMSSZ and YYYYMMDDHHMMSSZ, while the other functions allows others.